pkg/kubelet/util/pluginwatcher/plugin_watcher.go

```diff
@@ -158,10 +160,44 @@ func (w *Watcher) invokeRegistrationCallbackAtHandler(ctx context.Context, clien
 	}); err != nil {
 		return fmt.Errorf("failed to send registration status at socket %s, err: %v", socketPath, err)
 	}
+
 	chanForAckOfNotification <- true
 	return nil
 }
 
+// Handle filesystem notify event.
+func (w *Watcher) handlefsnotifyEvent(event fsnotify.Event) error {
+	if event.Op&fsnotify.Create != fsnotify.Create {
+		return nil
+	}
+
+	fi, err := os.Stat(event.Name)
+	if err != nil {
+		return fmt.Errorf("stat file %s failed: %v", event.Name, err)
+	}
+
+	if !fi.IsDir() {
+		err := w.registerPlugin(event.Name)
+		if err != nil {
+			return fmt.Errorf("Plugin %s registration failed with error: %v", event.Name, err)
+		}
+	} else {
+		if filepath.Base(event.Name)[0] == '.' {
+			return nil
+		}
+
+		if err := w.fsWatcher.Add(event.Name); err != nil {
+			return fmt.Errorf("failed to watch %s, err: %v", event.Name, err)
+		}
+
+		if err := w.traversePluginDir(event.Name); err != nil {
```

Comment:
Shouldn't TraversePluginDir always start at the root of `PluginsDir` ?  Otherwise you risk missing/not scanning dir added at level above `event.Name` no ?

Reply:
Thanks for review, let me try to explain why it will not miss dir,
TraversePluginDir() will be called to traverse root dir during Watcher.Start(),
once Watcher started, it will been called when a directory been created under root (in any depth).

come specific to your question that: if there is a directory above the level of `event.Name` been added, then a `CREATE` event will be generated by fsnotify for that dir above the level of 'event.Name' and it will be handled.